src/api/c++/z3++.h

```diff
@@ -2847,7 +2967,13 @@ namespace z3 {
             object::operator=(o);
             return *this;
         }
-        ~optimize() { Z3_optimize_dec_ref(ctx(), m_opt); }
+        optimize& operator=(optimize && o) noexcept {
+            object::operator=(std::forward<object>(o));
+            m_opt = o.m_opt;
+            o.m_opt = nullptr;
+            return *this;
+        }
+        ~optimize() { if(m_opt != nullptr) Z3_optimize_dec_ref(ctx(), m_opt); }
```

Comment:
convention used elsewhere (unfortunately we don't have a way to enforce conventions): "if (", not "if("

Reply:
I'll fix it. If it matters `clang-format` is pretty easy to set up if you want linting / convention enforcement. `cpp-check` as well.